src/main/java/org/wildfly/security/pem/Pem.java

```diff
@@ -69,18 +72,25 @@
             // BEGIN string is 11 chars long
             throw log.malformedPemContent(matcher.start() + 11);
         }
-        if (! pemContent.contentEquals("-----")) {
+        if (! pemContent.limitedTo(5).contentEquals("-----")) {
             throw log.malformedPemContent(pemContent.offset());
         }
-        final ByteIterator byteIterator = pemContent.delimitedBy('-').base64Decode();
+
+        String encodedKey = pemContent.delimitedBy('-').drainToString();
+
+        encodedKey = encodedKey.replaceAll("\r\n", "");
+        encodedKey = encodedKey.replaceAll("\n", "");
+        encodedKey = encodedKey.trim();
+
+        final ByteIterator byteIterator = ByteIterator.ofBytes(encodedKey.getBytes()).base64Decode();
```

Comment:
If we can get ``skipSpaces()``, for me is fine.

Do you want a separate PR for that or may I append another commit to this one ?